fapi-client/src/main/scala/com/gu/facia/api/utils/Backfill.scala

```diff
@@ -0,0 +1,63 @@
+package com.gu.facia.api.utils
+
+import com.gu.contentapi.client.GuardianContentClient
+import com.gu.facia.api.{FAPI, Response}
+import com.gu.facia.api.contentapi.ContentApi
+import com.gu.facia.api.contentapi.ContentApi._
+import com.gu.facia.api.models.{FaciaContent, CollectionConfig, CuratedContent}
+import com.gu.facia.client.ApiClient
+import com.gu.facia.client.models.{CollectionJson, Backfill, TrailMetaData}
+
+import scala.concurrent.ExecutionContext
+
+object BackfillContent {
+  def resolveFromConfig(collectionConfig: CollectionConfig): BackfillResolver = {
+    collectionConfig.backfill match {
+      case Some(backfill: Backfill) => backfill.`type` match {
+        case "capi" => CapiBackfill(backfill.query, collectionConfig)
+        case "collection" => CollectionBackfill(backfill.query)
+        case _ => throw new Error(s"Invalid backfill type ${backfill.`type`}")
```

Comment:
Are you sure you want an `Exception` here? Why not `EmptyBackfill`?
If so, write a quick unit test for it to show that's what is expected.